src/types.rs

```diff
@@ -1,10 +1,49 @@
+use std::collections::HashMap;
+
+use actix_web::http::Method;
+use actix_web::web::Bytes;
+use anyhow::Result;
 use dashmap::DashMap;
 use pyo3::prelude::*;
 
+#[pyclass]
 #[derive(Debug, Clone)]
-pub enum PyFunction {
-    CoRoutine(Py<PyAny>),
-    SyncFunction(Py<PyAny>),
+pub struct FunctionInfo {
+    pub handler: Py<PyAny>,
+    pub is_async: bool,
+    pub number_of_params: u8,
+}
+
+#[pymethods]
+impl FunctionInfo {
+    #[new]
+    pub fn new(handler: Py<PyAny>, is_async: bool, number_of_params: u8) -> Self {
+        Self {
+            handler,
+            is_async,
+            number_of_params,
+        }
+    }
+}
+
+#[derive(Default)]
+pub struct Request {
+    pub queries: HashMap<String, String>,
+    pub headers: HashMap<String, String>,
+    pub method: Method,
+    pub params: HashMap<String, String>,
+    pub body: Bytes,
+}
```

Comment:
@AntoineRR what `method` does it give to the request object on `Request:default()`?

Reply:
From actix_web:
```
impl Default for Method {
    #[inline]
    fn default() -> Method {
        Method::GET
    }
}
```
The default method is a GET then. It shouldn't matter for the const router, I just wanted a request to use the `execute_http_function` method, but it shouldn't be used in the case of the const router